include/api.php

```diff
@@ -1535,31 +1537,18 @@ function api_search($type)
 	} elseif (!empty($_REQUEST['count'])) {
 		$count = $_REQUEST['count'];
 	}
-
+	
 	$since_id = defaults($_REQUEST, 'since_id', 0);
 	$max_id = defaults($_REQUEST, 'max_id', 0);
 	$page = (!empty($_REQUEST['page']) ? $_REQUEST['page'] - 1 : 0);
 	$start = $page * $count;
 	$params = ['order' => ['id' => true], 'limit' => [$start, $count]];
-
-	if (!empty($_REQUEST['q'])) {
-		$condition = [
-			"`id` > ? 
-			AND (`uid` = 0 OR (`uid` = ? AND NOT `global`))
-			AND `body` LIKE CONCAT('%',?,'%')",
-			$since_id, api_user(), $_REQUEST['q']];
-		if ($max_id > 0) {
-			$condition[0] .= " AND `id` <= ?";
-			$condition[] = $max_id;
-		}
-	} elseif (!empty($_REQUEST['friendica_tag'])) {
-		$condition = [
-			"`oid` > ?
+	if (preg_match('/^\#([^#]+)/', $searchTerm, $matches) === 1 && isset($matches[1])) {
```

Comment:
There shouldn't be any white space in the hashtag (or punctuation) or else it would be a full-text search. Accents are allowed though.

Examples:
- `#foo` : Tag search
- `#foo bar`: Full-text search.

Reply:
@MrPetovan  
is this okay?
preg_match('/^#(\w+)$/', $searchTerm, $matches)

Reply:
Why not simply looking for a ```#``` as first parameter instead of a complicated regexp?

Reply:
Because `There shouldn't be any white space in the hashtag (or punctuation) or else it would be a full-text search.`.